File: lighthouse/src/main/java/com/automation/center/lighthouse/model/TestSuite.java
package com.automation.center.lighthouse.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Entity(name = "test_suite")
public final class TestSuite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "start_date")
    private LocalDateTime startDate;
    @Column(name = "end_date")
    private LocalDateTime endDate;
    @OneToMany(mappedBy = "testSuite")
    @ToString.Exclude
    private List<ResultHistory> resultHistories;
}

File: lighthouse/src/main/java/com/automation/center/lighthouse/model/TestSuite.kt
package com.automation.center.lighthouse.model

import jakarta.persistence.*
import lombok.Getter
import lombok.RequiredArgsConstructor
import lombok.Setter
import lombok.ToString

@Getter
@Setter
@ToString
@RequiredArgsConstructor
@Entity(name = "test_suite")
class TestSuite {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var name: String? = null
    var description: String? = null
    var cron: String? = null

    @ManyToMany
    @JoinTable(
        name = "test_suite_metric",
        joinColumns = [JoinColumn(name = "test_suite_id")],
        inverseJoinColumns = [JoinColumn(name = "metric_id")]
    )
    val metrics: List<Metric> = ArrayList()

    @OneToMany
    @JoinColumn(name = "test_suite_id")
    val urls: List<SuiteUrl> = ArrayList()
}
